cadc-util/src/main/java/ca/nrc/cadc/auth/Authenticator.java

```diff
@@ -72,22 +72,39 @@
 import javax.security.auth.Subject;
 
 /**
- * Subject modification interface.
+ * Subject validation and modification interface.
  *
  * @author pdowler
  */
 public interface Authenticator {
+    
+    /**
+     * If necessary, validate any principals in the subject.  Some principals,
+     * such as X500Principal, do not require validation as that is done with
+     * TLS.  Others, such as cookies and tokens, need to be validated.
+     * Failed validation must result in an AccessControlException
```

Comment:
javadoc and throws inconsistent

The expected usage needs to be documented and this seems like the place. My understanding is that the incoming subject could contain an **unvalidated** AuthorizationTokenPrincipal and a typical validate would do something  like:
* parse/validate the AuthorizationTokenPrincipal and put the resulting AuthorizationToken into the pub credentials
* replace the AuthorizationTokenPrincipal with an HttpPrincipal if a username can be extracted, or wait for augment(...), or do do nothing... but should probably say that an authenticated user must have a principal so just removing it would be bad

That would be the same for a CookiePrincipal. 
Some related comments below in AuthorizationTokenPrincipal and AuthorizationToken

Reply:
typo in the javadoc "AuthoricationToken"